lib/widgets/app.dart

```diff
@@ -268,17 +266,18 @@ class ChooseAccountPage extends StatelessWidget {
         // The default trailing padding with M3 is 24px. Decrease by 12 because
         // IconButton (the "…" button) comes with 12px padding on all sides.
         contentPadding: const EdgeInsetsDirectional.only(start: 16, end: 12),
-        onTap: () => Navigator.push(context,
-          HomePage.buildRoute(accountId: accountId))));
+        onTap: () => HomePage.navigate(context, accountId: accountId)));
   }
 
   @override
   Widget build(BuildContext context) {
     final zulipLocalizations = ZulipLocalizations.of(context);
     assert(!PerAccountStoreWidget.debugExistsOf(context));
     final globalStore = GlobalStoreWidget.of(context);
+    final hasBackButton = ModalRoute.of(context)?.impliesAppBarDismissal ?? false;
```

Comment:
This line is pretty mysterious — let's give it a one- or two-line comment pointing at how you determined this was the right way to compute whether there'll be a back button.